File: app/src/main/java/com/example/shibeapplication/repositories/Repository.java
package com.example.shibeapplication.repositories;

import com.example.shibeapplication.retrofit.RetrofitInstance;
import com.example.shibeapplication.retrofit.ShibeInterface;

import java.util.List;

import retrofit2.Call;
import retrofit2.Retrofit;

public class Repository implements ShibeInterface {

    private Repository() {

    }

    private static class RepositoryHolder {
        public static final Repository INSTANCE = new Repository();
    }

    public static Repository getInstance() {
        return RepositoryHolder.INSTANCE;
    }


    @Override
    public Call<List<String>> getShibes(int count) {

        return RetrofitInstance.getInstance()
                .create(ShibeInterface.class)
                .getShibes(count);

    }
}

File: app/src/main/java/com/example/shibeapplication/model/repositories/Repository.kt
package com.example.shibeapplication.model.repositories

import com.example.shibeapplication.model.retrofit.ShibeInterface
import retrofit2.Call
import retrofit2.Retrofit
import javax.inject.Inject

class Repository @Inject constructor(private val retrofit: Retrofit) : ShibeInterface {

    override fun getShibes(count: Int): Call<List<String>> {
        return retrofit
                .create(ShibeInterface::class.java)
                .getShibes(count)
    }

}